src/test/java/core/basesyntax/service/ActionStrategyImplTest.java

```diff
@@ -0,0 +1,86 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.action.ActionHandler;
+import core.basesyntax.service.action.BalanceAction;
+import core.basesyntax.service.action.PurchaseAction;
+import core.basesyntax.service.action.ReturnAction;
+import core.basesyntax.service.action.SupplyAction;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ActionStrategyImplTest {
+    private static final Map<Operation, ActionHandler> actionHandlerMap = Map.of(
+            Operation.BALANCE, new BalanceAction(),
+            Operation.PURCHASE, new PurchaseAction(),
+            Operation.RETURN, new ReturnAction(),
+            Operation.SUPPLY, new SupplyAction()
+    );
+    private final ActionStrategy strategy = new ActionStrategyImpl(actionHandlerMap);
+
+    @Test
+    void checkBalanceActionGet_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction(Operation.BALANCE, "banana", 20);
+        assertEquals(BalanceAction.class, strategy.get(fruitTransaction.getOperation()).getClass());
+    }
+
+    @Test
+    void checkBalanceActionGetWithWrongAmount_NotOk() {
+        FruitTransaction fruitTransaction =
+                new FruitTransaction(Operation.BALANCE, "banana", -20);
+        assertThrows(RuntimeException.class, () ->
+                strategy.get(fruitTransaction.getOperation())
+                        .count(fruitTransaction.getFruit(), fruitTransaction.getQuantity()));
+    }
+
+    @Test
+    void checkPurchaseActionGet_Ok() {
+        FruitTransaction fruitTransaction =
+                new FruitTransaction(Operation.PURCHASE, "banana", 20);
+        assertEquals(PurchaseAction.class,
+                strategy.get(fruitTransaction.getOperation()).getClass());
```

Comment:
```suggestion
        FruitTransaction fruitTransaction =
                new FruitTransaction(Operation.PURCHASE, "banana", 20);
                Action action = fruitTransaction.getOperation()
                
        assertEquals(PurchaseAction.class,
                action.getClass());
```

Reply:
better to separate method call in this case to make test more readable and divide it into 3 parts (precondition, execution, check)